### DISCUSSION TITLE
[Feature Request] Add buy button for Pro games/accounts

### DISCUSSION THREAD

**Discussion by 5H00T4** [ChristopherKlay/StadiaEnhanced#55] [Ideas & Requests]
So I recently wanted to use the coupon that I get to buy a game before my Pro Subscription runs out, but Google has actually disabled that functionality, i.e. the buy button on Accounts with an active Pro Subscription.
Meaning I couldn't use the coupon and the game would vanish from my library once my sub would run out.

So I did some sleuthing on another account without Pro and the button was there of course.
I inspected the element and after some trial and error I could display the button by simply copying the correct JS code.
It seems like there is no backend check, and this could theoretically work for everyone everywhere.

I've seen this discussion a lot of times on reddit, could really help folks out. 👍

**Comment by ChristopherKlay** [comment]
It's definitely possible to add a purchase option to Pro games, but it comes with a couple of issues;

It's only possible to effectively rebuild the button itself, that triggers a javascript action on Stadias end. It isn't possible to check if a game is actually purchasable (i.e. games only on Pro), if the user already owns a version of the game, or things like the actual price (including deals and such). Thus i could definitely add a "Purchase Pro Game" option, but it would just be a simple button without much labeling (outside of the games name and such) that brings up the purchase overview (which shows the price and other aspects of the purchase).

There's also a legal issue; Offering a work-around for otherwise blocked purchases is definitely something that could result in issues for both the project and people buying games via it. It's insanely easy to spot that people can suddenly buy games they "own" already and i would need to talk to Google first.

That being said; I'll send a little pigeon with a letter their way. 😉

**Edit:** Update.
I've got into contact with someone on Googles end and while they can't provide any info about the issue (which is to be expected, since i wasn't in contact with the team managing for example the store itself) they will forward the issue with a reference to this project including my contact details. Let's hope this at the very least gets the stone rolling.

**Reply by 5H00T4** [reply]
You're right, I also thought about the implications.

That being said, they only block purchases for Pro users, as I could buy the game from another account.
I would assume there are backend checks in place to prevent users from buying stuff twice or whatever.

Not what their reasons are, but seeing as they actively block people from buying Pro games they must have thought about this use case. ;)
